ang/civiawards/award-creation/directives/review-panels/review-panels.directive.js

```diff
@@ -305,16 +410,21 @@
      * @returns {Promise} promise
      */
     function prepareRelationshipsForSave () {
-      return _.map($scope.currentReviewPanel.relationships, function (relation) {
-        var isAToB = relation.type.indexOf('a_b') !== -1;
-        var relationshipTypeId = relation.type.substr(0, relation.type.indexOf('_'));
-
-        return {
-          is_a_to_b: isAToB ? '1' : '0',
-          relationship_type_id: relationshipTypeId,
-          contact_id: getSelect2Value(relation.contacts)
-        };
-      });
+      return _.chain($scope.currentReviewPanel.relationships)
+        .filter(function (relation) {
+          return relation.type.length !== 0;
```

Comment:
why was this filter introduced? How come a relationship type can be empty?

Reply:
Because we always have an empty relationship added so that user can enter data, but is should not be saved if its empty. So we remove it before saving.

![2020-03-19 at 11 00 AM](https://user-images.githubusercontent.com/5058867/77034788-d5d76a80-69d0-11ea-81ff-ff4731457b41.jpg)